pkg/process_manager.go

```diff
@@ -18,46 +26,183 @@ const (
 )
 
 type ProcessStatus struct {
-	// ExitStatus is only valid if `State` is `EXITED`.
-	ExitStatus int32
-	State      ProcessState
+	// ExitCode is only valid if `State` is `EXITED`.
+	ExitCode int
+	State    ProcessState
 }
 
 type processStream struct {
 	// TODO
 }
 
-func (*processStream) Read(p []byte) (n int, err error) {
+func (s *processStream) Read(p []byte) (n int, err error) {
 	panic("TODO")
 }
 
+type process struct {
+	cancel         context.CancelFunc
+	statusLock     sync.RWMutex
+	status         ProcessStatus
+	terminated     atomic.Bool
+	terminatedCond *sync.Cond
+}
+
+func (s *process) markTerminated() {
+	s.terminated.Store(true)
+	s.terminatedCond.Broadcast()
```

Comment:
This works, but I feel like it would be a fair bit simpler to use a channel close (one concurrency primitive) instead of an atomic bool, mutex, and condvar (three concurrency primitives).

Reply:
> What isn't clear to me is how/why to replace the RWMutex with channels.

Sorry, I wasn't clear in my communication. You'll need to keep the mutex for other shared access, you just won't need to use it with a condvar anymore.

> since future attempts to read will immediately return the zero value, which I didn't know before.

Yes, exactly. Closing the channel will unblock anyone currently waiting to read on it (like broadcasting to a cond var), AND cause all future reads to instantly return (eliminating the need to track the closed state).